fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java

```diff
@@ -646,8 +656,14 @@ private GlobalStateMgr(boolean isCkptGlobalState) {
         this.auditEventProcessor = new AuditEventProcessor(this.pluginMgr);
         this.analyzeManager = new AnalyzeManager();
 
-        if (Config.use_staros) {
+        if (Config.run_mode.equalsIgnoreCase(RunMode.SHARED_DATA.name())) {
+            runMode = RunMode.SHARED_DATA;
             this.starOSAgent = new StarOSAgent();
+        } else if (Config.run_mode.equalsIgnoreCase(RunMode.SHARED_NOTHING.name())) {
+            runMode = RunMode.SHARED_NOTHING;
```

Comment:
You'd better implement this string to Enum in the RunMode class, and then it can be reused.